src/components/Watchlist/MovieCardFetcher.tsx

```diff
@@ -0,0 +1,40 @@
+import { useEffect, useState } from "react";
+import { MovieCard, MovieCardProps } from "../pages/MovieSearchPage/MovieCard";
+
+import { Grid } from "@mantine/core";
+import { transformMovieData } from "../pages/MovieSearchPage/MovieCardList";
+
+export default function MovieCardFetcher({ id }) {
+  const [movieData, setMovieData]: [MovieCardProps | undefined, Function] =
+    useState();
+
+  const getMovieByID = (id: number) => {
```

Comment:
Create issue for improvment, unify api calls either fetch .then.catch or try catch